### PR TITLE
Orc-272: [C++] Minor porting changes.

### PR DESCRIPTION
**Pull request by rip-nsk**

### CODE REVIEW

**Comment by xndai** [comment]
LGTM

File: c++/src/Int128.cc [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <iomanip>
 #include <iostream>
 #include <sstream>
+#include <algorithm> // std::min
```
Comment (wgtmac) [inline c++/src/Int128.cc]: nit: you may want to sort them alphabetically.

**Comment by omalley** [comment]
This patch works on MacOS, but fails on CentOS 6. In general, for this kind of change, I like to run the docker/run-all.sh on the patch.  I get:

```
[ 42%] Building CXX object c++/src/CMakeFiles/orc.dir/io/InputStream.cc.o
In file included from /root/orc/c++/src/io/InputStream.cc:19:
/root/orc/c++/include/orc/Exceptions.hh:31: error: expected ';' before 'noexcept'
/root/orc/c++/include/orc/Exceptions.hh:41: error: expected ';' before 'noexcept'
/root/orc/c++/include/orc/Exceptions.hh:51: error: expected ';' before 'noexcept'
make[2]: *** [c++/src/CMakeFiles/orc.dir/io/InputStream.cc.o] Error 1
```

**Comment by rip-nsk** [comment]
I don't think it is related to the proposed changes since io/InputStream.cc (and any headers) are not touched.

**Comment by omalley** [comment]
Ok, now that I have travis happy again, let me test this one out.

**Pull request closed without merging**